libpromises/cmdb.c

```diff
@@ -194,6 +201,93 @@ static bool ReadCMDBVars(EvalContext *ctx, JsonElement *vars)
     return true;
 }
 
+static StringSet *GetTagsFromJsonTags(const char *item_type,
+                                      const char *key,
+                                      const JsonElement *json_tags,
+                                      const char *default_tag)
+{
+    StringSet *tags = NULL;
+    if (json_tags != NULL)
+    {
+        if ((JsonGetType(json_tags) != JSON_TYPE_ARRAY) ||
+            (!JsonArrayContainsOnlyPrimitives((JsonElement*) json_tags)))
+        {
+            Log(LOG_LEVEL_ERR,
+                "Invalid json_tags information for %s '%s' in CMDB data:"
+                " must be a JSON array of strings",
```

Comment:
You don't check whether or not the primitives are strings?

Reply:
Not sure if I need to care, all primitives are strings internally and all can/will be converted to strings by `JsonArrayToStringSet()` below. If user wants numbers as tags, they can do so.

Reply:
Okey, thanks for clarification :smile:

Reply:
Should we maybe include an edge-case in the acceptance test where one of these other primitives are used?

Reply:
I don't think that's necessary. We have a test for `JsonArrayToStringSet()` with a non-string primitive.

Reply:
Ah, okey. Nice!